pkg/controller/clusterclaim/clusterclaim_controller.go

```diff
@@ -253,11 +253,28 @@ func (r *ReconcileClusterClaim) Reconcile(ctx context.Context, request reconcile
 
 	cd := &hivev1.ClusterDeployment{}
 	switch err := r.Get(context.Background(), client.ObjectKey{Namespace: clusterName, Name: clusterName}, cd); {
-	case apierrors.IsNotFound(err):
-		return r.reconcileForDeletedCluster(claim, logger)
-	case err != nil:
+	case err != nil && !apierrors.IsNotFound(err):
 		logger.Log(controllerutils.LogLevel(err), "error getting ClusterDeployment")
 		return reconcile.Result{}, err
+	case apierrors.IsNotFound(err) || cd.DeletionTimestamp != nil || controllerutils.IsClaimedClusterMarkedForRemoval(cd):
```

Comment:
small nit:
this switch is no longer readable imo. 
just add the new conditions to separate if after the switch.

Reply:
I tried refactoring this a couple of ways, ended up pulling the complicated bit into a helper func. Hope this suits.

My mental notes:
We want to blow up if:
- The error is not nil AND
- The error is something other than `IsNotFound`
We want to `reconcileForDeletedCluster` if:
- The error `IsNotFound`; OR
- There is no error AND (the CD has a deletion timestamp OR the removal annotation)
Continue otherwise, i.e. if:
- There is no error AND
- The CD has no deletion timestamp AND
- The CD has no removal annotation